explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+use the signed exchange which header-integrity is sha256-AAA if available in the
+cache.
+
+When the sha256-AAA signed exchange exists in the cache but the sha256-BBB
+signed exchange doesn’t exists, the UA which supports WebP format MUST ignore
+the sha256-AAA signed exchange and fetch the original URL. Otherwise this can be
+used for sending tracking ID.
+
+The **alternate** link also can have
+[variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+attributes to support content negotiation for recursive prefetch.
+
+- In **outer** HTTP response of article.html.sxg from feed.example:
+    ```
+    Link: <https://feed.example/publisher.example/image_jpeg.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          anchor="https://publisher.example/image";
+    Link: <https://feed.example/publisher.example/image_webp.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          anchor="https://publisher.example/image";
+    ```
+- In **inner** response header of article.html.sxg:
+    ```
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          header-integrity="sha256-AAA"
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          header-integrity="sha256-BBB"
+    Link: <https://publisher.example/image>; rel=preload; as=image;
+    ```
+If a UA supports WebP, the UA should prefetch **image_webp.sxg** which content
+is WebP format. Otherwise the UA should prefetch **image_jpeg.sxg** which
+content is JPEG format.
+
+## Security and Privacy Considerations
+- The publishers can know whether the referrer page has prefetched the signed
+exchange subresources or not by checking the resource timing information. But
+this only exposes 1 bit information (= yes or no) because UAs can use the cached
+signed exchange only if the required signed exchanges are all available.
+- The UA can use the prefetched signed exchange subresources, only when they
+were prefetched in the referrer page. This is intended to avoid leaking the
+prefetching state to succeeding pages.
+- The UA check the header integrity value, so the distributor of the subresource
+signed exchange can’t inject arbitrary resources to the publisher’s page. This
+prevents distributors from sending tracking IDs to the publisher’s page.
+- The UA need to check both
+[the request URL](https://wicg.github.io/webpackage/loading.html#exchange-request-url)
+and the header integrity value of the signed exchange to avoid
```

Comment:
"The UA needs to check that the signed request URL matches the preload link and not just that the header-integrity value matches, since the header-integrity hash doesn't cover the request URL."